editoast/src/views/train_schedule/projection.rs

```diff
@@ -277,40 +274,40 @@ async fn project_path(
     let signal_updates = signal_updates?;
 
     // 3. Store the projection in the cache (using pipeline)
-    let trains_hash_values: HashMap<_, _> = trains
+    let trains_hash_values: HashMap<_, _> = trains_details
         .iter()
-        .map(|t| t.id)
+        .map(|t| t.train_id)
```

Comment:
Am I correct if I say that, should we use a `TrainId` type instead of a `i64` everywhere we need identifiers, we would not have had this bug? Is this something we want to investigate?

Reply:
It's not a type issue, so even with a `TrainId` type, we would have the bug.